pkg/cleaner/cleaner.go

```diff
@@ -26,16 +26,18 @@ func Run(gitHome string, nsLister k8s.NamespaceWatcher, fs sys.FS) error {
 		log.Printf("unable to get watch events (%s)", err)
 	}
 	for {
-		event := <-watcher.ResultChan()
-		if event.Type == "DELETED" {
-			switch event.Object.(type) {
-			case (*api.Namespace):
-				namespace := event.Object.(*api.Namespace)
-				appToDelete := gitHome + "/" + namespace.ObjectMeta.Name + dotGitSuffix
-				if err := fs.RemoveAll(appToDelete); err != nil {
-					log.Printf("Cleaner error removing deleted app %s (%s)", appToDelete, err)
+		select {
+		case event := <-watcher.ResultChan():
```

Comment:
How does the logic here change from the above `event := <-watcher.ResultChan()` wait?

Also, as a defensive measure, can you check for channel close here and exit the loop if so?